### PR TITLE
Enable smoke tests for web

### PR DESCRIPTION
**Pull request by jeanp413**
Related https://github.com/gitpod-io/gitpod/issues/3065

Run this to run the smoke tests for web:
```
yarn smoketest-no-compile --web --verbose --headless --electronArgs="--disable-dev-shm-usage --use-gl=swiftshader"
```

### CODE REVIEW

File: src/vs/server/browser/workbench/workbench.ts [hunk 1]
```diff
@@ -296,6 +296,15 @@ class WindowIndicator implements IWindowIndicator {
 
 
 (function () {
+	// Find config by checking for DOM
+	const configElement = document.getElementById('vscode-workbench-web-configuration');
+	const configElementAttribute = configElement ? configElement.getAttribute('data-settings') : undefined;
+	if (!configElement || !configElementAttribute) {
+		throw new Error('Missing web configuration element');
```
Comment (akosyakov) [inline src/vs/server/browser/workbench/workbench.ts:303]: it should be optional

File: src/vs/server/node/server.main.ts [hunk 2]
```diff
@@ -199,9 +204,29 @@ async function serveFile(logService: ILogService, req: http.IncomingMessage, res
...
-	res.end(errorMessage);
+async function handleRoot(req: http.IncomingMessage, resp: http.ServerResponse, entryPointPath: string, environmentService: INativeEnvironmentService) {
+	if (!req.headers.host) {
+		return serveError(req, resp, 400, 'Bad request.');
+	}
+
+	const host = req.headers.host;
+
+	const workbenchConfig = {
+		remoteAuthority: host,
```
Comment (akosyakov) [inline src/vs/server/node/server.main.ts:215]: we don't need it

File: test/smoke/src/areas/explorer/explorer.test.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (akosyakov) [inline test/smoke/src/areas/explorer/explorer.test.ts:1]: we should use our copyrights for new files:

```
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Gitpod. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
```

File: src/vs/server/node/server.main.ts [hunk 4]
```diff
@@ -558,6 +583,8 @@ export async function main(options: IServerOptions): Promise<void> {
 		const requestService = accessor.get(IRequestService);
 		channelServer.registerChannel('request', new RequestChannel(requestService));
 
+		const environmentService = accessor.get(INativeEnvironmentService);
```
Comment (akosyakov) [inline src/vs/server/node/server.main.ts:586]: i think it is already declared above

**Comment by akosyakov** [comment]
@jeanp413 let's have a look at it in the clean workspace, i failed to run them

**Comment by jeanp413** [comment]
@akosyakov I've seen this before, I think it's related to how gitpod caches the node_modules.
Try running this first:
```
yarn install --force
```

**Review by akosyakov — APPROVED** [review]
lgtm

i will remove leeway later

**Pull request merged**